elements/lisk-transactions/src/3_vote_transaction.ts

```diff
@@ -70,12 +89,63 @@ export class VoteTransaction extends BaseTransaction {
 		const tx = (typeof rawTransaction === 'object' && rawTransaction !== null
 			? rawTransaction
 			: {}) as Partial<TransactionJSON>;
-		this.asset = (tx.asset || {}) as VoteAsset;
+		if (tx.asset) {
+			const rawAsset = tx.asset as RawAsset;
+			this.asset = {
+				votes: rawAsset.votes,
+				recipientId: rawAsset.recipientId || this.senderId,
+				amount: new BigNum(
+					isValidNumber(rawAsset.amount) ? rawAsset.amount : '0',
+				),
+			};
+		} else {
+			// tslint:disable-next-line no-object-literal-type-assertion
+			this.asset = {} as VoteAsset;
+		}
 		this.containsUniqueData = true;
 	}
 
-	protected assetToBytes(): Buffer {
-		return Buffer.from(this.asset.votes.join(''), 'utf8');
+	public assetToJSON(): object {
+		return {
+			votes: this.asset.votes,
+			amount: this.asset.amount.toString(),
+			recipientId: this.asset.recipientId,
+		};
+	}
+
+	public sign(passphrase: string, secondPassphrase?: string): void {
+		super.sign(passphrase, secondPassphrase);
+		this.asset.recipientId = this.senderId;
+	}
+
+	protected getBasicBytes(): Buffer {
```

Comment:
As the other comment here it would be nice to have a comment about the reasons we're overriding this method.